src/hooks/useFetchImages.ts

```diff
@@ -1,71 +1,78 @@
 import { useEffect, useState } from 'react'
+import { useQuery } from '@tanstack/react-query'
 import unixToDate from '../utilities/date-lib/unixToDate'
 import dateToUnix from '../utilities/date-lib/dateToUnix'
 import unixNow from '../utilities/date-lib/unixNow'
 import ky from 'ky'
 
-const useFetchImages = (needMoreImages: boolean) => {
-  const [isLoading, setIsLoading] = useState(false)
-  const [images, setImages] = useState<Image[]>([])
+type UseFetchImagesResult = {
+  isLoading: boolean
+  error?: Error
+  data: Image[]
+}
+
+const useFetchImages = (needMoreImages: boolean): UseFetchImagesResult => {
```

Comment:
Now we can narrow the type of your hook to return the same type useQuery returns including `isFetching`
```suggestion
const useFetchImages = (needMoreImages: boolean): UseQueryResult<Image[], Error> => {
```

Reply:
It seems I could only declare the type of data property which would be `Image[]`, so: 
`const useFetchImages = (needMoreImages: boolean): UseQueryResult<Image[]> => { ... }`

When trying to explicitly type `boolean`, or `Error` as you did in the second argument, it says I can't return the whole query as a result. 

I still have access to the isFetching and other properties in the homepage, so figured I should be okay. Thoughts?